3di | Contributing to ros2 documentation (#6966)
ros2/ros2_documentation: source/The-ROS2-Project/Contributing/Documentation/Creating-or-updating-documentation.rst

If you are creating a new article, decide on the content type for the article before you start.

For more information about the docs source, tools, and workflow to use when making your updates, see :doc:`../Contributing-to-documentation`.

Building the site locally
^^^^^^^^^^^^^^^^^^^^^^^^^

1 Setting up the documentation tools
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Set up the following prerequisites to build the docs site locally:

#. Create a `venv <https://docs.python.org/3/library/venv.html>`__ to build the documentation:

   .. code-block:: console

      $ python3 -m venv ros2doc  # create venv
      $ source ros2doc/bin/activate  # activate venv

#. Install the requirements located in the ``requirements.txt`` file:

   .. tabs::

      .. group-tab:: Linux

         .. code-block:: console

            $ pip install -r requirements.txt -c constraints.txt

      .. group-tab:: macOS

         .. code-block:: console

            $ pip install -r requirements.txt -c constraints.txt

      .. group-tab:: Windows

         .. code-block:: console

            $ python -m pip install -r requirements.txt -c constraints.txt

> #. Make sure the ``dot`` command is available, so that Sphinx is able to generate diagrams:

Reviewer: I am unclear what ``dot`` means in this context.  The lines below indicate that you are simply installing graphviz. Wouldn't it be clearer to say as much?

Reviewer 2: The key aspect is that the `dot` command is available; installing `graphiz` is a means to that end.

Reviewer 2: ```suggestion
#. Make sure the ``dot`` command, typically provided as part of ``graphviz``, is available, so that Sphinx is able to generate diagrams:
```

> #. Make sure the ``dot`` command, typically provided as part of ``graphviz``, is available, so that Sphinx is able to generate diagrams:

Reviewer: ```suggestion
#. Sphinx generates diagrams using the ``graphviz`` library so make sure that it is installed and  available.
```

This was still really unclear to me. It could be that I'm not familiar with the `dot` command (I've never used and it is likely  most other readers haven't either).

Author: Thanks - I made a tiny whitespace and punctuation fix to this and added under a new commit.

.. code-block:: text

   empy
   jupyter
   lets
   ws

To include custom corrections that ``codespell`` should apply, you can add them to the `codespell_dictionary <https://github.com/ros2/ros2_documentation/blob/{REPOS_FILE_BRANCH}/codespell_dictionary.txt>`_ file as follows:

.. code-block:: text

   amnet->ament
   colcn->colcon
   rosabg->rosbag
   rosdistroy->rosdistro

To check the dictionaries, run the following command:

.. code-block:: console

   $ make check-dictionaries

This command checks the blank lines and leading/trailing spaces in the dictionaries.

If the check-dictionaries command complains about the dictionaries, run the following command:

.. code-block:: console

   $ make sort-dictionaries

This command automatically modifies the dictionaries if any issues are found.

4 Checking for broken links
~~~~~~~~~~~~~~~~~~~~~~~~~~~

To check for broken links on the site, run the following command:

.. code-block:: console

   $ make linkcheck

This will check the entire site for broken links, and output the results to the screen and ``build/linkcheck``.

> 5 Building the site for one branch

Reviewer: ```suggestion
5 Building the site for the active branch
```

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> To build the site for just this branch:

Reviewer: ```suggestion
To build the site for just the current active branch:
```
